src/rebar3_hex_owner.erl

```diff
@@ -37,20 +37,48 @@ do(State) ->
     end.
 
 handle_command(State, Repo) ->
-    case rebar_state:command_args(State) of
-        ["add", Package, UsernameOrEmail] ->
-            add(Repo, Package, UsernameOrEmail, State),
+    case command_args(State) of
+        {"add", Package, UsernameOrEmail} ->
+            Config = rebar3_hex_utils:hex_config_for(write, Repo),
+            add(Config, Package, UsernameOrEmail, State),
             {ok, State};
-        ["remove", Package, UsernameOrEmail] ->
-            remove(Repo, Package, UsernameOrEmail, State),
+        {"remove", Package, UsernameOrEmail} ->
+            Config = rebar3_hex_utils:hex_config_for(write, Repo),
+            remove(Config, Package, UsernameOrEmail, State),
             {ok, State};
-        ["list", Package] ->
-            list(Repo, Package, State),
+        {"list", Package} ->
+            Config = rebar3_hex_utils:hex_config_for(read, Repo),
+            list(Config, Package, State),
             {ok, State};
         _Command ->
             ?PRV_ERROR(bad_command)
     end.
 
+command_args(State) ->
```

Comment:
I'm not so sure this a good idea... if support this here then we have to support it everywhere. Maybe it's just better to have *better* error messages and require the user always append the `-r` switch and argument to the end.